File: Ruby/linked _list.rb
class Node
    attr_accessor :value,:nextNode

    def initialize(value)
        @value = value
        @nextNode = nil
    end

    def linkNext(node)
        @nextNode = node
    end

    def getNext
        @nextNode
    end
end

class LinkedList
    attr_accessor :head,:size

    def initialize()
        @head = nil
        @size = 0
    end

    def addHead(value)
        if @head == nil
            @head = Node.new(value)
        else
            temp = @head
            newHead = Node.new(value)
            newHead.linkNext(temp)
            @head = newHead
        end
        @size+=1
    end

end




obj1 = LinkedList.new()
obj1.addHead('Forrest')

obj1.addHead('Dragon ball')
obj1.addHead('perriyo')



puts obj1.head.nextNode.nextNode.value

File: Python/linked_list.py
class Node:
    def __init__(self,value):
        self.value = value
        self.nextNode = None
    
    def linkNext(self,node):
        self.nextNode = node

    def getNext(self):
        return self.nextNode
    
class LinkedList:
    def __init__(self):
        self.head = None
        self.size = 0
    

    def addHead(self,value):
        if self.head == None:
            self.head = Node(value)
        else:
            temp = self.head
            newHead = Node(value)
            newHead.linkNext(temp)
            self.head = newHead
        
        self.size+=1
    



obj1 = LinkedList()
obj1.addHead("2")
obj1.addHead("eee")


print(str(obj1.head.value))




